include/caffe/layer.hpp

```diff
@@ -316,6 +317,21 @@ class Layer {
     param_propagate_down_[param_id] = value;
   }
 
+  bool IsForwardPassed() const {
```

Comment:
Now please look at net.cpp:

+  for (int i = start; i <= end; ++i) {
+    layers_[i]->ForwardPassed(true);
+  }

Here layers[i] is a pointer to the *base* class Layer. How would we call ForwardPassed here?